src/Microsoft.DotNet.Interactive/Parsing/SubmissionParser.cs

```diff
@@ -78,19 +78,27 @@ private IReadOnlyList<KernelCommand> SplitSubmission(KernelCommand originalComma
 
                         if (parseResult.Errors.Any())
                         {
-                            commands.Clear();
-                            commands.Add(
-                                new AnonymousKernelCommand((kernelCommand, context) =>
-                                {
-                                    var message =
-                                        string.Join(Environment.NewLine,
-                                                    parseResult.Errors
-                                                               .Select(e => e.ToString()));
-
-                                    context.Fail(message: message);
-                                    return Task.CompletedTask;
-                                }, parent: originalCommand.Parent));
+                            if (parseResult.Errors.All(e => e.SymbolResult == null) && directiveNode is ActionDirectiveNode actionDirectiveNode)
```

Comment:
I'm not sure what `SymbolResult == null` is supposed to indicate or whether it will behave consistently the way you expect.

Reply:
that is my question, an unknown directive is producing an error without any symbols. I am actually looking for a way to discriminate by something used in the wrong way (other tests are covering scenarios with wrong options and args) and something that is unknown

Reply:
I suspect a heuristic approach will be needed, and a bunch of tests covering various kinds of parsing errors.